core/extension/gdextension_interface.h

```diff
@@ -256,6 +256,7 @@ typedef struct {
 
 typedef const GDExtensionPropertyInfo *(*GDExtensionClassGetPropertyList)(GDExtensionClassInstancePtr p_instance, uint32_t *r_count);
 typedef void (*GDExtensionClassFreePropertyList)(GDExtensionClassInstancePtr p_instance, const GDExtensionPropertyInfo *p_list);
+typedef void (*GDExtensionClassFreePropertyList2)(GDExtensionClassInstancePtr p_instance, const GDExtensionPropertyInfo *p_list, uint32_t p_count);
```

Comment:
Raul pointed this out on the godot-cpp PR - the `const` here is actually incorrect (because we're freeing the memory this points at), and we're working around it in godot-cpp with a `const_cast`. This would also be a good time to fix that as well!

```suggestion
typedef void (*GDExtensionClassFreePropertyList2)(GDExtensionClassInstancePtr p_instance, GDExtensionPropertyInfo *p_list, uint32_t p_count);
```

Reply:
Or, hrm... Thinking about this a little bit more, I'm not sure `const` is incorrect. If we remove the `const`, then Godot will have to do the `const_cast` in order to call this function!

`get_property_list` returns `const GDExtensionPropertyInfo *` because we don't want Godot to modify this data, it's "owned" by the GDExtension. But since it will have a `const GDExtensionPropertyInfo *`, it can't call `free_property_list` with a `GDExtensionPropertyInfo *` without casting it. But, arguably, it'd be better for the GDExtension to do the cast because it's the one that "owns" this data.

So, on 2nd thought, maybe `const` is correct, and we need to update the comment in godot-cpp about it being incorrect?

Reply:
isn't it then `get_property_list` that's wrong? i think it returns a const pointer, but that pointer is later gonna be freed so it shouldn't be a const pointer?

Reply:
It's getting returned to Godot, though, and it's not Godot that's going to free it. It's going to pass it back to the GDExtension to do the freeing. From the perspective of Godot, I feel like a const pointer is correct.

Reply:
from the perspective of godot, isn't this effectively the same as a normal alloc + free (except that you're given initialized memory here)? like sure the gdextension is who actually performs the logic to do the freeing, but the same is true for calling the `free` function for a memory allocator. the memory allocator is actually the one freeing the memory but we're still responsible for triggering that logic by calling `free`.

Reply:
To my understanding, "able to free" doesn't imply "must not be `const`".
This works fine in C++:
```cpp
const int* p = new int(30);
delete p;
```

Now, C [`free()`](https://en.cppreference.com/w/c/memory/free) does take a non-const pointer, but that's likely due to its age. 
There's also some discussion [on StackOverflow](https://stackoverflow.com/q/2819535).